sourcing-new/src/main/scala/ch/epfl/bluebrain/nexus/sourcingnew/projections/JdbcProjection.scala

```diff
@@ -0,0 +1,86 @@
+package ch.epfl.bluebrain.nexus.sourcingnew.projections
+
+import akka.persistence.query.Offset
+import cats.effect.{Async, ContextShift}
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.sourcingnew.projections.ProjectionProgress.NoProgress
+import ch.epfl.bluebrain.nexus.sourcingnew.projections.instances._
+import com.typesafe.config.Config
+import doobie.implicits._
+import doobie.util.transactor.Transactor
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder}
+
+final case class JdbcConfig(driver: String,
+                            host: String,
+                            port: Int,
+                            database: String,
+                            username: String,
+                            password: String) {
+  def url: String = s"jdbc:postgresql://$host:$port/$database?stringtype=unspecified"
+}
+
+class JdbcProjection[F[_]: ContextShift: Async,
+                     A: Encoder: Decoder](jdbcConfig: JdbcConfig,
+                                          journalConfig: Config)
+  extends Projection[F, A] {
+
+  private val progressTable: String       = journalConfig.getString("projection-progress-table")
+  private val failuresTable: String       = journalConfig.getString("projection-failures-table")
+
+  private val xa = Transactor.fromDriverManager[F](
+    jdbcConfig.driver,
+    jdbcConfig.url,
+    jdbcConfig.username,
+    jdbcConfig.password,
+  )
+
+  /**
+    * Records progress against a projection identifier.
+    *
+    * @param id       the projection identifier
+    * @param progress the offset to record
+    * @return a future () value
+    */
+  override def recordProgress(id: String, progress: ProjectionProgress): F[Unit] =
+    sql"UPDATE $progressTable SET progress = ${progress.asJson.noSpaces} WHERE projection_id = $id".
```

Comment:
This is not going to work for postgres. For cassandra it works because if the row does not exist it will just create it.

For Postgres we need an Upsert as follows:

```scala
val jsonString = progress.asJson.noSpaces
sql"""
INSERT INTO $progressTable (projection_id, progress) VALUES('$id', '$jsonString') 
ON CONFLICT (projection_id) DO  UPDATE SET progress = '$jsonString';
"""
```

Reply:
I see you have already found out :)